components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,116 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:PrizePool
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Opponent = require('Module:Opponent')
+local Table = require('Module:Table')
+
+local PrizePool = Class.new(function(self, ...) self:init(...) end)
+
+-- TODO: Extract Enum handling to its own Module
+local prizeTypes = Table.map({
+    'USD', 'LOCAL_CURRENCY', 'POINTS', 'QUALIFIES', 'FREETEXT'
+}, function(value, name) return name, value end)
+
+-- TODO: Add possiblity to add custom prizeData
+local prizeData = {
+    [prizeTypes.USD] = {row = 'usdprize'},
+    [prizeTypes.LOCAL_CURRENCY] = {header = 'localcurrency', row = 'localprize'},
+    [prizeTypes.POINTS] = {header = 'points', row = 'points'},
+    [prizeTypes.QUALIFIES] = {header = 'qualifies', row = 'qualified'},
+    [prizeTypes.FREETEXT] = {header = 'freetext', row = 'freetext'},
+}
+
+function PrizePool:init(args)
+    self.args = args
+    self.pagename = mw.title.getCurrentTitle().text
+    self.args.type = Json.parseIfString(self.args.type)
+    self.widgetInjector = nil
+    self.lpdbInjector = nil
+
+    if not self.args.type then
+        return error('Please provide a type!')
+    elseif type(self.args.type) ~= 'table' or not self.args.type.type then
+        return error('Could not parse type!')
+    elseif not Opponent.isType(self.args.type.type) then
+        return error('Not a valid type!')
+    end
+
+    self.type = self.args.type.type
+
+    self.options = self:_readOptions(args)
+    self.prizes = self:_readPrizes(args)
+    if self.options.showUSD then
+        self:addPrize('USD', 1)
+    end
+    self.placements = {} -- TODO
+end
+
+function PrizePool:create()
+	-- TODO
+	return ''
+end
+
+function PrizePool:_readOptions(args)
```

Comment:
should be either overwritable or expandable
e.g. sc2 would like to add stuff like import limit etc.

Reply:
The latter